docs/howto/manage-storage.md

```diff
@@ -164,7 +96,107 @@ The storage will not be available immediately after that call - the charm should
+
+Next, access the storage in `_update_configuration`. Don't hard-code the location of the storage. Instead, get the location from [`ContainerMeta.mounts`](ops.ContainerMeta.mounts) for the workload container:
+
+```python
+def _update_configuration(self, event: ops.EventBase):
+    """Update the workload configuration."""
+    cache = self.model.storages["cache"]
+    if not cache:
+        # The storage-attached event hasn't happened yet.
+        logger.info("Storage is not yet ready.")
+        return
+    # Storage is available. Write some data to the cache.
+    cache_in_container = self.meta.containers["web-service"].mounts["cache"]
+    cache_root = pathops.ContainerPath(
+        cache_in_container.location,
+        container=self.unit.get_container("web-service"),
+    )
+    try:
+        (cache_root / "data.json").write_text("...")
+    except ops.pebble.ConnectionError:
+        # Pebble isn't ready yet. Return and wait for the pebble-ready event.
+        logger.info("Pebble is not yet ready.")
+        return
+```
+
+This example uses {external+charmlibs:ref}`charmlibs-pathops <charmlibs-pathops>` to write data to the storage in the workload container.
+
+Instead of directly writing data to the storage, your charm could put the storage location in a configuration file. For example:
+
+```python
+    ...
+    # Storage is available. Pass the storage location to the workload.
+    cache_in_container = self.meta.containers["web-service"].mounts["cache"]
+    config_path = pathops.ContainerPath(
+        "/etc/my-app/cache-path.config",
+        container=self.unit.get_container("web-service"),
+    )
+    try:
+        config_path.write_text(cache_in_container.location)
+    except ops.pebble.ConnectionError:
+        ...
+    # Ask the workload to reload its configuration.
+```
+
+## Handle storage detaching
+
+In the `src/charm.py` file, in the `__init__` function of your charm, set up an observer for the detaching event associated with your storage and pair that with an event handler. For example:
+
+```
+self.framework.observe(self.on.cache_storage_detaching, self._on_storage_detaching)
+```
+
+> See more: [](ops.StorageDetachingEvent)
+
+Now, in the body of the charm definition, define the event handler, or adjust an existing holistic one. For example, to warn users that data won't be cached:
+
+```
+def _on_storage_detaching(self, event: ops.StorageDetachingEvent):
+    """Handle the storage being detached."""
+    self.unit.status = ops.ActiveStatus("Caching disabled; provide storage to boost performance)
+```
+
+> Examples: [MySQL handling cluster management](https://github.com/canonical/mysql-k8s-operator/blob/4c575b478b7ae2a28b09dde9cade2d3370dd4db6/src/charm.py#L823), [MongoDB updating the set before storage is removed](https://github.com/canonical/mongodb-operator/blob/b33d036173f47c68823e08a9f03189dc534d38dc/src/charm.py#L596)
+
+## Write unit tests
```

Comment:
I think we could have a better test example here, where the interaction with the storage is in the charm (maybe actually the content from the examples above) rather than faking it all in the context manager. But that could also easily be left for a future improvement.